chapter05/Task_5_2.py

```diff
@@ -0,0 +1,117 @@
+# Task 5.2
+# Program about hero and his specifications
+#  User can customize hero
+
+specification = {
+    'Strength': 0,
+    'health': 0,
+    'wisdom': 0,
+    'agility': 0}
+
+points = 30
+
+print('''
+        Hello and Welcome
+    Today you customize your hero
+
+    Let's start:''')
+
+while True:
+    choice = input('''
+1. Start to customize hero
+2. Edit strength
+3. Edit health
+4. Edit wisdom
+5. Edit agility
+
+Remember, you can only distribute {} points\n'''.format(points))
+
+    if choice == '1':
+        for item in specification:
+            while points >= 0:
```

Comment:
It's easier to read the loop when the "main" flow is within the loop and not hidden under 3rd "else" statement.

Instead consider the following:

```
while points >= 0:
    how_many = ...
    if points - how_many < 0 or points - how_many > 30:
        print("...")
        continue

     if specification[item] + how_many < 0:
          print("...")
          continue

     # Here we have a clear "main" path where user inputs a good value 
     specifications[item] += how_many
     points -= how_many
     break

```